### DISCUSSION TITLE
Feedback on cperl-mode and completion

### DISCUSSION THREAD

**Discussion by smonff** [HaraldJoerg/cperl-mode#12] [General]
I don't have a lot of luck with completion and `cperl-mode`. 

Honestly I am not sure it is `cperl-mode` fault but maybe only my lack of motivation. I usually use no completion at all, or not  something *"smart"* at all. I recently re-tried to set up some interesting tools (`company`, `plsense`, `lsp`, etc.) and it was a total fail. Some interesting results came with `plsense` and `company`, but I never managed to have the completion fired up while typing, it was only triggered by a specific keybinding.

If some persons have any clue or full setups to share for Emacs based modern Perl completion configs, I am interested. Most of those stuff are 10 years old at least, or I don't even know about them.

Note: I spend something like 5 years on Prelude Emacs what is not a very Perl friendly environment (the defaults are always messing up badly with my config and I never managed to fix that). With Doom Emacs, it is a bit different: there is no Perl layer at all, so it is just the Emacs defaults. I kind of like it, but it also makes more complicated to find other people to share experiences about it.

**Comment by HaraldJoerg** [comment]
I rarely use completion (I find it often distracts me with way too many alternatives), but the combination of plsense and auto-complete gives me completion while typing (after activating auto-complete-mode, of course).
For the language server protocol, the Perl counterpart [Perl::LanguageServer](https://github.com/richterger/Perl-LanguageServer) is fairly recent, but integration with lsp or eglot seems to be missing, or underdocumented.

**Reply by smonff** [reply]
Thanks for the `auto-complete` reminder. Used it many years ago but then forgot about it. 

Other things you mention, for sure, under-documented or not stable.
